horace_core/sqw/@sqw/sqw.m

```diff
@@ -168,3 +140,46 @@
 
 end
 
+end  % function
+
+
+% -----------------------------------------------------------------------------
+function w = from_file_name(file_name, dnd_type, class_type, varargin)
+    if ~dnd_type
```

Comment:
There are three different argument parsers handling the varargininput to the get_data, get_sqw and PixelData constructor calls. 
It may be possible to simplify these are as part of the SQW refactor work, different approaches to argument handling makes the code harder to follow.